tests/clpy_tests/opencl_tests/ultima_tests/test_half.py

```diff
@@ -10,21 +10,22 @@
 class TestUltimaHalfTrick(unittest.TestCase):
 
     def test_type_half(self):
-        x = '''
+        x = clpy.backend.ultima.exec_ultima('', '#include <cupy/carray.hpp>') + ('''
 __clpy__half f() 
 {
-    __clpy__half a;
-    return (__clpy__half)(42.F);
+    __clpy__half a;constructor___clpy__half___left_paren____clpy__half_float__right_paren__(&a, 42.F);
+    return a;
 }
-'''
+''')[1:]
         y = clpy.backend.ultima.exec_ultima(
             '''
             half f(){
-              half a;
-              return static_cast<half>(42.f);
+              half a = 42.f;
```

Comment:
Can you address the caller of such constructor (float2half)? Because custom kernel won't accept to use half directly, we can call `convert_*()` directly in clpy, right?